fabric-command-api-v2/src/client/java/net/fabricmc/fabric/mixin/command/client/ClientPlayNetworkHandlerMixin.java

```diff
@@ -72,6 +72,11 @@ private void onOnCommandTree(CommandTreeS2CPacket packet, CallbackInfo info) {
 		ClientCommandInternals.addCommands((CommandDispatcher) commandDispatcher, (FabricClientCommandSource) commandSource);
 	}
 
+	@Inject(method = "onCommandTree", at = @At("HEAD"))
+	private void setLastReceivedCommandsPacket(CommandTreeS2CPacket packet, CallbackInfo ci) {
+		ClientCommandInternals.setLastReceivedCommandsPacket(packet);
```

Comment:
Is this thread safe? Can this be called on the network thread at the same time someone calls refreshCommandCompletions on some other thread?